Dynatrace/dynatrace-operator
# Version status uses endpoints instead of image manifest

## Description

Version status (OA,CodeModule,AG) doesn't use image manifest to get version of the image. The logic depends on the registry:
- for Tenant registry -> the response of the two endpoints (`.../latest/metainfo`) are used
- for Public registry -> no changes
- for custom image -> we do not detect the version for custom image anymore. For this reason downgrade check is also removed for custom image case. Additionally no verification of provided URI is performed,

`registry.GetImageVersion` call has been also removed from the Synthetic updater. Temporarily `Version` set to `latest`.

## How can this be tested?
1) Use unit tests `make go/test` to verify if tenant endpoints are used to get image version. 

2) Check if `*.Status.Version` field is set properly.

tenant registry - latest version on your tenant will be reported:
```
spec:
  activeGate:
    capabilities:
      - kubernetes-monitoring
  oneAgent:
    cloudNativeFullStack: {}
---
Status:
  Active Gate:
    Image ID:              <tenant>.dev.dynatracelabs.com/linux/activegate:latest
    Source:                tenant-registry
    Version:               1.285.0.20240123-020204
  Code Modules:
    Source:                tenant-registry
    Version:               1.285.0.20240122-141707
  One Agent:
    Image ID:              <tenant>.dev.dynatracelabs.com/linux/oneagent:latest
    Source:                tenant-registry
    Version:               1.285.0.20240122-141707
```
custom version - use version from the `tenant-registry` step and then do downgrade to a previous version, it should be allowed
```
spec:
  activeGate:
    capabilities:
      - kubernetes-monitoring
  oneAgent:
    cloudNativeFullStack:
      version: "1.285.0.20240122-141707"
---
Status:
  Active Gate:
    Image ID:              <tenant>.dev.dynatracelabs.com/linux/activegate:latest
    Source:                tenant-registry
    Version:               1.285.0.20240123-020204
  Code Modules:
    Source:                tenant-registry
    Version:               1.285.0.20240122-141707
  One Agent:
    Image ID:              <tenant>.dev.dynatracelabs.com/linux/oneagent:1.285.0
    Source:                custom-version
    Version:               1.285.0.20240122-141707
```
custom image -  use latest versions from the `public.ecr.aws` repo and then do downgrade to a previous versions, it should be allowed. Reported `Version` is always set to `custom-image`.
```
spec:
  activeGate:
    capabilities:
      - kubernetes-monitoring
    image: "public.ecr.aws/dynatrace/dynatrace-activegate:1.279.110.20231121-222817"
  oneAgent:
    cloudNativeFullStack:
      image: "public.ecr.aws/dynatrace/dynatrace-oneagent:1.281.150.20240122-144834"
---
Status:
  Active Gate:
    Image ID:              public.ecr.aws/dynatrace/dynatrace-activegate:1.279.110.20231121-222817
    Source:                custom-image
    Version:               custom-image
  Code Modules:
    Source:                tenant-registry
    Version:               1.285.0.20240122-141707
  One Agent:
    Image ID:              public.ecr.aws/dynatrace/dynatrace-oneagent:1.281.150.20240122-144834
    Source:                custom-image
    Version:               custom-image
```

## Checklist
